app/jobs/concerns/failure_notifiable_job.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module FailureNotifiableJob
```

Comment:
This might be a bit too generic, maybe `DeliveryFailureNotifiableJob`?

Reply:
DeliveryFailure sounds more like failure to deliver a notification

Reply:
FailedDeliveryNotifiable?

Reply:
ok